rbd/snapshot_nautilus.go

```diff
@@ -65,6 +65,64 @@ func (image *Image) GetParentInfo(pool, name, snapname []byte) error {
 	return nil
 }
 
+// ImageSpec represents the image information.
+type ImageSpec struct {
+	ImageName string
+	PoolName  string
+}
+
+// SnapSpec represents the snapshot infomation.
+type SnapSpec struct {
+	ID       uint64
+	SnapName string
+}
+
+// ParentInfo represents the parent image and the parent snapshot information.
+type ParentInfo struct {
+	ImgSpec ImageSpec
+	SnpSpec SnapSpec
+}
+
+// GetParent looks for the parent of the image and returns the parent image
+// information which includes the image name, the pool name and
+// the snapshot information.
+// If parent image is not present it returns without error.
```

Comment:
I did that because of two reasons:

1. Request from cephcsi: https://github.com/ceph/ceph-csi/pull/1432#discussion_r475660140
2. Some of the cli functions already do that https://github.com/ceph/ceph/blob/957a5cfbf5e4a830aafe4c98f8febcea07d3e846/src/cls/rbd/cls_rbd.cc#L1605

Reply:
OK, I still think it needs changing though. We should not hide the errors coming back from the api. CSI and other tools can just as easily check for an ENOENT equivalent flag. If you're concerned about making it less difficult to check for, why not make the new error value you added an exported value?

Reply:
I think then better to return getError() only because cephcsi can check against ErrNotFound